Gradle 7 compatibility

## Steps to Reproduce

<!-- Please tell us exactly how to reproduce the problem you are running into. -->

1.  Update your gradle wrapper to get 7.0-rc-1.
2. Try to build

## Logs
```
FAILURE: Build failed with an exception.

* What went wrong:
Some problems were found with the configuration of task ':app:compileFlutterBuildDebug' (type 'FlutterTask').
  - Type 'FlutterTask' property 'assets' is missing an input or output annotation.

    Reason: A property without annotation isn't considered during up-to-date checking.

    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.

    Please refer to https://docs.gradle.org/7.0-rc-1/userguide/validation_problems.html#missing_annotation for more details about this problem.
  - Type 'FlutterTask' property 'assetsDirectory' is missing an input or output annotation.

    Reason: A property without annotation isn't considered during up-to-date checking.

    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.

    Please refer to https://docs.gradle.org/7.0-rc-1/userguide/validation_problems.html#missing_annotation for more details about this problem.
  - Type 'FlutterTask' property 'buildMode' is missing an input or output annotation.

    Reason: A property without annotation isn't considered during up-to-date checking.

    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.

    Please refer to https://docs.gradle.org/7.0-rc-1/userguide/validation_problems.html#missing_annotation for more details about this problem.
  - Type 'FlutterTask' property 'fastStart' doesn't have a configured value.

    Reason: This property isn't marked as optional and no value has been configured.

    Possible solutions:
      1. Assign a value to 'fastStart'.
      2. Mark property 'fastStart' as optional.

    Please refer to https://docs.gradle.org/7.0-rc-1/userguide/validation_problems.html#value_not_set for more details about this problem.
  - Type 'FlutterTask' property 'flutterExecutable' is missing an input or output annotation.

    Reason: A property without annotation isn't considered during up-to-date checking.

    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.

    Please refer to https://docs.gradle.org/7.0-rc-1/userguide/validation_problems.html#missing_annotation for more details about this problem.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 5s
Exception: Gradle task assembleDebug failed with exit code 1

```

<details>
<summary>Logs</summary>

<!-- If possible, paste the output of running `flutter doctor -v` here. -->

```
[✓] Flutter (Channel stable, 2.0.3, on Microsoft Windows [Version 10.0.19042.870], locale en-IN)
    • Flutter version 2.0.3 at E:\Software\IDEs\Android\flutter
    • Framework revision 4d7946a68d (8 days ago), 2021-03-18 17:24:33 -0700
    • Engine revision 3459eb2436
    • Dart version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at E:\Software\IDEs\Android\android-sdk
    • Platform android-S, build-tools 30.0.3
    • ANDROID_HOME = E:\Software\IDEs\Android\android-sdk
    • Java binary at: E:\Software\IDEs\Android\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = E:\Software\Browsers\Chrome\Chrome Stable x64 Launcher.exe

[✓] Android Studio (version 4.1.0)
    • Android Studio at E:\Software\IDEs\Android\android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] Connected device (2 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 10 (API 29) (emulator)
    • Chrome (web)                       • chrome        • web-javascript • unknown

• No issues found!
```

</details>

--- a/packages/flutter_tools/gradle/flutter.gradle
+++ b/packages/flutter_tools/gradle/flutter.gradle
@@ -953,18 +953,21 @@ class FlutterExtension {
 }
 
 abstract class BaseFlutterTask extends DefaultTask {
+    @Internal
     File flutterRoot
+    @Internal
     File flutterExecutable
+    @Input
     String buildMode
     @Optional @Input
     String localEngine
     @Optional @Input
     String localEngineSrcPath
-    @Input
+    @Optional @Input
     Boolean fastStart
     @Input
     String targetPath
-    @Optional
+    @Optional @Internal
     Boolean verbose
     @Optional @Input
     String[] fileSystemRoots
@@ -974,7 +977,9 @@ abstract class BaseFlutterTask extends DefaultTask {
     Boolean trackWidgetCreation
     @Optional @Input
     List<String> targetPlatformValues
+    @Internal
     File sourceDir
+    @Internal
     File intermediateDir
     @Optional @Input
     String extraFrontEndOptions
@@ -992,6 +997,7 @@ abstract class BaseFlutterTask extends DefaultTask {
     String bundleSkSLPath
     @Optional @Input
     String codeSizeDirectory;
+    @Optional @Input
     String performanceMeasurementFile;
     @Optional @Input
     Boolean deferredComponents
@@ -1091,10 +1097,12 @@ class FlutterTask extends BaseFlutterTask {
         return intermediateDir
     }
 
+    @Internal
     String getAssetsDirectory() {
         return "${outputDirectory}/flutter_assets"
     }
 
+    @Internal
     CopySpec getAssets() {
         return project.copySpec {
             from "${intermediateDir}"
@@ -1102,6 +1110,7 @@ class FlutterTask extends BaseFlutterTask {
         }
     }
 
+    @Internal
     CopySpec getSnapshots() {
         return project.copySpec {
             from "${intermediateDir}"

